src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py

```diff
@@ -8,512 +8,417 @@
-            total = result["TotalRecords"]
-            if "Extras" in result:
-                gLogger.info(result["Extras"])
-                extra = result["Extras"]
-                timestamp = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M [UTC]")
-                callback = {"success": "true", "result": callback, "total": total, "extra": extra, "date": timestamp}
-            else:
-                callback = {"success": "true", "result": callback, "total": total, "date": None}
-
-            gLogger.info("\033[0;31m PRODUCTION SUBMIT REQUEST: \033[0m %s" % (datetime.datetime.utcnow() - pagestart))
-        self.finish(json.dumps(callback))
-
-    ################################################################################
-    @asyncGen
-    def web_action(self):
-        transid = int(self.get_argument("id"))
-        if self.get_argument("data_kind") == "getLoggingInfo":
-            callback = yield self.threadTask(self.__getLoggingInfo, transid)
-        elif self.get_argument("data_kind") == "fileStatus":
-            callback = yield self.threadTask(self.__transformationFileStatus, transid)
-        elif self.get_argument("data_kind") == "fileProcessed":
-            callback = yield self.threadTask(self.__fileRetry, transid, "proc")
-        elif self.get_argument("data_kind") == "fileNotProcessed":
-            callback = yield self.threadTask(self.__fileRetry, transid, "not")
-        elif self.get_argument("data_kind") == "fileAllProcessed":
-            callback = yield self.threadTask(self.__fileRetry, transid, "all")
-        elif self.get_argument("data_kind") == "dataQuery":
-            callback = yield self.threadTask(self.__dataQuery, transid)
-        elif self.get_argument("data_kind") == "additionalParams":
-            callback = yield self.threadTask(self.__additionalParams, transid)
-        elif self.get_argument("data_kind") == "transformationDetail":
-            callback = yield self.threadTask(self.__transformationDetail, transid)
-        elif self.get_argument("data_kind") == "extend":
-            callback = yield self.threadTask(self.__extendTransformation, transid)
-        elif self.get_argument("data_kind") == "workflowxml":
-            callback = yield self.threadTask(self.__workflowxml, transid)
-        else:
-            callback = {"success": "false", "error": "Action is unknown!!!"}
-        self.finish(callback)
+        result = tsClient.getTransformationSummaryWeb(result, self.globalSort, start, limit)
+        if not result["OK"]:
+            return json.dumps({"success": "false", "error": result["Message"]})
+
+        data = result["Value"]
+
+        if "TotalRecords" not in data:
+            return json.dumps({"success": "false", "result": "", "error": "Data structure is corrupted"})
+
+        if (total := data["TotalRecords"]) < 1:
+            return json.dumps({"success": "false", "result": "", "error": "There were no data matching your selection"})
```

Comment:
Is the `json.dumps` needed? Or could the dictionary be returned directly?

Reply:
No, `BaseRequestHandler` already has encoder. Fixed